modules/openapi-generator/src/test/resources/3_0/python-experimental/petstore-with-fake-endpoints-models-for-testing-with-http-signature.yaml

```diff
@@ -1539,6 +1543,22 @@ components:
             type: object
             additionalProperties:
               type: string
+        anytype_1:
+          type: object
+        anytype_2: {}
+        anytype_3:
+          type: object
+          properties: {}
+        map_with_additional_properties:
+          type: object
+          additionalProperties: true
+        map_without_additional_properties:
+          type: object
+          additionalProperties: false
+        map_string:
+          type: object
+          additionalProperties:
+            type: string
```

Comment:
These property names are being added, but their values disagree with their descriptions.
Can you change them to be consistent so our users aren't confused?

- anytype_1 results in `{str: (bool, date, datetime, dict, float, int, list, str, none_type)}` how about naming it map_of_anytype1
- anytype_2 results in `bool, date, datetime, dict, float, int, list, str, none_type` so how about naming it anytype1
- anytype_3 results in `{str: (bool, date, datetime, dict, float, int, list, str, none_type)}` so how about naming it map_of_anytype2
- map_with_additional_properties results in `{str: (bool, date, datetime, dict, float, int, list, str, none_type)}` so how about naming it map_of_anytype3
- map_without_additional_properties results in `bool, date, datetime, dict, float, int, list, str` so how about renaming it anytype2

map_without_additional_properties lacks none_type in its resultant data_type? Is that correct or incorrect?

Reply:
> These property names are being added, but their values disagree with their descriptions.
> Can you change them to be consistent so our users aren't confused?
> 
> * anytype_1 results in `{str: (bool, date, datetime, dict, float, int, list, str, none_type)}` how about naming it map_of_anytype_1

done

> * anytype_2 results in `bool, date, datetime, dict, float, int, list, str, none_type` so how about naming it anytype_1

done

> * anytype_3 results in `{str: (bool, date, datetime, dict, float, int, list, str, none_type)}` so how about naming it map_of_anytype_2

done

> * map_with_additional_properties results in `{str: (bool, date, datetime, dict, float, int, list, str, none_type)}` so how about naming it map_of_anytype_3

done

> * map_without_additional_properties results in `bool, date, datetime, dict, float, int, list, str` so how about renaming it anytype_2

I think it should be name `empty_map`. Since `additionalProperties` is False, undeclared properties are not allowed; and there are no declared properties, hence no properties are allowed. I think the allowed types are set to `bool, date, datetime, dict, float, int, list, str` because there are no restrictions on the type, but since there are no declared and undeclared properties, nothing will be allowed.

> 
> map_without_additional_properties lacks none_type in its resultant data_type? Is that correct or incorrect? Isn't that incorrect?
> 
> * Shouldn't map_without_additional_properties be a map where only certain specific key property names are allowed because additionalProperties is False?

No, I wanted to have an empty map.

> * Do we need to convert that into a a new separate object type CodegenModel which becomes a Python ModelNormal to enforce that behavior?

What is `that`?

Reply:
I have some more questions here:

map_with_additional_properties results in {str: (bool, date, datetime, dict, float, int, list, str, none_type)} so how about naming it map_of_anytype_3
done.
Update: actually I don't think it's a good suggestion. We would lose the fact that the map allows undeclared properties using multiple variants. I will name it map_with_undeclared_properties_anytype_1/2/3.
There are three properties allowing undeclared properties.

1. What does multiple variants mean?
2. What is the difference between undeclared properties and additionalProperties: True?

The example that you want to name empty_map.
I am okay naming this empty_map, but the python code is not meeting your intention of only allowing an empty map here.

3. How do you propose having the python code only allow empty map for this property?
There are 2 solutions that I see:
a) when this condition is seen (property schema of type object with no properties and additionalProperties: False) in Java we need to set the maxItems for this property to 0. That will only allow empty dictionaries here.
b) create a new CodegenModel form the inline schema with no properties and additionalProperties: False, this is a complicated solution

Reply:
Separate issue #6400 is tracking this. Marking resolved.